src/manifolds/Segre.jl

```diff
@@ -0,0 +1,473 @@
+@doc raw"""
+    Segre{𝔽,V} <: AbstractManifold{𝔽}
+
+The Segre manifold
+````math
+    \mathcal{S} = \operatorname{Seg}(𝔽^{n_1} \times \dots \times 𝔽^{n_d})
+````
+is the set of rank-one tensors in ``𝔽^{n_1} \otimes \dots \otimes 𝔽^{n_d}``.
+
+When ``𝔽 = ℝ``, the Segre manifold is represented as
+````math
+    \mathcal{S} \sim ℝ^{+} \times S^{n_1 - 1} \times \dots \times S^{n_d - 1}.
+````
+This is a local diffeomorphism, and the metric is a locally a [warped product](https://en.wikipedia.org/wiki/Warped_product) of ``ℝ^{+}`` and ``S^{n_1 - 1} \times \dots \times S^{n_d - 1}``. The tuple ``(n_1, \dots, n_d)`` is called the _valence_ of the manifold.
+
+The geometry is summarized in [JacobssonSwijsenVandervekenVannieuwenhoven:2024](@cite).
+
+# Constructor
+    Segre(valence::NTuple{V, Int}; field::AbstractNumbers=ℝ)
+
+Generate a valence `V` Segre manifold.
+"""
+struct Segre{𝔽,V} <: AbstractManifold{𝔽} end
+
+function Segre(valence::NTuple{V,Int}; field::AbstractNumbers=ℝ) where {V}
+    return Segre{field,valence}()
+end
+
+valence(::Segre{𝔽,V}) where {𝔽,V} = V
+ndims(::Segre{𝔽,V}) where {𝔽,V} = length(V)
```

Comment:
These two functions no where tested or used. Are they really necessary? Then they should be covered by tests. Since they are also not exported, they would for now only be internal functions anyways.

Reply:
I'll remove them!